SMelidoni/personal-finance-manager
# feature/logout-functionality

## Description of Changes

This PR introduces a secure logout functionality to the application. This change allows users to securely log out from both the backend and the frontend of the application. 

## Type of Change

- Bug Fix
- Breaking Change
- Documentation Update
- Other

## Describe the problem this PR solves and your solution.

Previously, users did not have the ability to log out of the application. This PR solves this issue by implementing a secure logout mechanism. The backend invalidates the JWT token by setting it in an HTTP-only, secure cookie with a past expiration date. On the frontend, the user's authentication state is removed from local state.

## Testing Steps
1. Login to the application using valid credentials.
2. In your browser's developer tools, you should be able to see this cookie under the "Cookies" tab in the "Application" or "Storage" panel.
3. Click the logout button.
4. Verify that the user is redirected to the login page.
5. Check the cookie has been removed/deleted.

## Screenshots

N/A

## Checklist
